lib/fetch/headers.js

```diff
@@ -456,15 +456,27 @@ class Headers {
     //    set with all the names of the headers in list.
     const names = [...this[kHeadersList]]
     const namesLength = names.length
-    if (namesLength <= 16) {
-      // Note: Use insertion sort for small arrays.
-      for (let i = 1, value, j = 0; i < namesLength; ++i) {
-        value = names[i]
-        for (j = i - 1; j >= 0; --j) {
-          if (names[j][0] <= value[0]) break
-          names[j + 1] = names[j]
+    if (namesLength <= 64) {
```

Comment:
Move this into it's own util file and use the same function for implementaiton and test

Reply:
also some comments please, as to why 64 was chosen

Reply:
I dont see any reason to even keep this check as i doubt there are real world cases where we hit more than 64 elements

Reply:
I considered that, but it's a harmless change that would be better off with a comment explaining it.

Reply:
+1 on util and comments

Does this has to do with pre-allocation?

Reply:
> also some comments please, as to why 64 was chosen

Fast up to 64? This number is because we have already verified that it is.

Reply:
> I dont see any reason to even keep this check as i doubt there are real world cases where we hit more than 64 elements

Insertion sort cannot be used in large arrays because the worst case is O(n^2).
V8 uses tim sort (an improved version of merge sort), which is slow because it is a stable sort. 
Tim sort in small arrays, binary insertion sort is used, so its fast-path is implemented in JS-land for optimization.
Use other algorithm? In this case, there are no duplicate keys, so unstable sort can be used.

Reply:
> Does this has to do with pre-allocation?

No.

Reply:
> Fast up to 64? This number is because we have already verified that it is. (generally speaking, speeds below 32 are said to be fast.)

yeah, so a comment would be helpful so someone can easily tell why it was chosen

Reply:
TIL